src/Microsoft.Diagnostics.DebugServices.Implementation/ServiceEvent.cs

```diff
@@ -11,42 +11,23 @@ namespace Microsoft.Diagnostics.DebugServices.Implementation
     /// </summary>
     public class ServiceEvent : IServiceEvent
     {
-        private class EventNode : LinkedListNode
+        private class EventNode : LinkedListNode, IDisposable
```

Comment:
Any reason we are keeping the custom linked list?

Reply:
Using the standard library LinkedListNode<T>/LinkedList<T> sucks.  It doesn't allow anyway for the "node" (LinkedListNode<T>) to remove itself from the list. Remove is on LinkedList<T> so you would have to pass the list head and save it in the node so the node can do the remove.  It isn't any better than using List<T>.

I really want to keep my LinkedListNode implementation.  It is simple, efficient, handles removal during Fire and works/is tested.